src/nbl/asset/interchange/CImageLoaderJPG.cpp

```diff
@@ -339,8 +339,12 @@ asset::SAssetBundle CImageLoaderJPG::loadAsset(system::IFile* _file, const asset
 	core::smart_refctd_ptr<ICPUImage> image = ICPUImage::create(std::move(imgInfo));
 	image->setBufferAndRegions(std::move(buffer), regions);
 
-	auto hash = contentHasher.finalizeSeq();
-	image->setContentHash(hash);
+	{
+		auto hash = contentHasher.finalizeSeq();
+		auto* buffer = image->getBuffer();
+		buffer->setContentHash(buffer->computeContentHash());
+		image->setContentHash(hash);
+	}
```

Comment:
>  decide if hash for an ICPUImage's buffer should be calculated, or invalidated

Here's an example code. 
If computing hash for an `ICPUImage`'s buffer is unnecessary, we can invalidate it instead.

Reply:
the hash of the buffer is unecessary indeed (ICPUImage never looks at the buffer data hash, because it has no bearing on the image data uniqueness).